getting_started_modules.md

```diff
@@ -0,0 +1,341 @@
+- `main.tf` contains the [resources](https://www.terraform.io/docs/configuration/resources.html) used.
+- `variables.tf` contains all variables used. 
+- `outputs.tf` defines any output variables (if relevant). 
+All files should be put in the root (same as the folder this README is in). 
+
+
+##### main.tf
+In our case the only thing our main.tf should contain is a resource that takes our nomad-job file and deploys it to Nomad. Below is an example of such a resource:
+
+```hcl-terraform
+resource "nomad_job" "countdash" {
+  jobspec = file("${path.module}/conf/nomad/countdash.hcl")
+  detach  = false
+}
+```
+
+`${path.module}` is the path to where our module is. `detach = false` tells terraform to wait for the service to be healthy in nomad before finishing. 
+[Documentation on the resource used](https://registry.terraform.io/providers/hashicorp/nomad/latest/docs/resources/job).
+
+
+##### variables.tf
+
+> :warning: In the first iteration we don't need any input variables.
+
+In this file you define any variables you would want to be [input variables](https://www.terraform.io/docs/configuration/variables.html) to your module. 
+If we are provisioning a postgres service, maybe we'd like a "Name of postgres database" variable as input, or "Number of servers to provision" if you are provisioning a cluster.  
+A variable is defined like below:
+
+```hcl-terraform
+variable "service_name" {
+  type        = string
+  description = "Minio service name"
+  default     = "minio"
+}
+```
+
+##### outputs.tf
+
+> :warning: In the first iteration we don't need any output variables.
+
+This files contains variables that will be available as [output variables](https://www.terraform.io/docs/configuration/outputs.html) when you use a module. 
+Below is first an example of how to define output-variables, then an example of how to use a module, and access their output variables.
+Defining output variables:
+
+ ```hcl-terraform
+output "nomad_job" {
+  value       = nomad_job.countdash
+  description = "The countdash nomad job object"
+}
+```
+
+> :bulb: Together inputs and outputs should create a very clear picture of how a module should be used. 
+>For example in our hive module we have clearly defined that it needs to have a postgres-address as an input. 
+>In our postgres module we have an output that is exactly that. In other words, we might need to import and setup a postgres-module before setting up our hive-module, so that we get a postgres-address to give our hive-module. 
+>Or, if we already have a postgres-address available, we could supply that instead. The goal is to clearly define the needs of a module, while at the same time making it flexible and generic (in the example of hive we give the user the ability to use any postgres they'd like). 
+>How to use variables and outputs will be shown later in [make the module more dynamic]().
```

Comment:
Missing link `[make the module more dynamic]()`

Reply:
fixed!